### ISSUE TITLE
LDAP Sync Error: Invalid DN Syntax

### ISSUE THREAD

**Issue by voegel916** [snipe/snipe-it#7519]
I've quite a bit of trouble with LDAP, but finally got it working last week and synced all of our employees into Snipe. Yesterday when I tried syncing to bring in some of our new employees, it started returning the error above. I've tried adding quotes, changing between caps/lowercase, changing the DNs, but it still gives the same error. Any advice?

#### Expected Behavior (or desired behavior if a feature request)

Perform LDAP User sync, new AD users are synced and added to the current list of people.

-----

#### Actual Behavior

Click Synchronize, page loads for a minute before returning the message (Whoops, looks like something went wrong."

-----

#### Please confirm you have done the following before posting your bug report:

- [X ] I have enabled debug mode 
- [X ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----

- Is this a fresh install or an upgrade? 
-- Fresh Install

- Version of Snipe-IT you're running 
-- v4.7.4

- Version of PHP you're running 
-- v7.2.19

- Version of MySQL/MariaDB you're running 
-- v15.1

- What OS and web server you're running Snipe-IT on 
-- Ubuntu 18.04, Apache

- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) 
-- install.sh

- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
-- ldap_search(): Search: Invalid DN syntax /var/www/snipe-it/app/Models/Ldap.php#265

- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
-- On the LDAP Sync page, using the synchronize button.

- Any errors that appear in your browser's error console.
-- Failed to load resource: the server responded with a status of 500 (Internal Server Error)

- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo.
-- LDAP sync not available in the demo.

- Include what you've done so far in the installation, and if you got any error messages along the way.
-- Fully installed, no error messages other than LDAP. 
- Indicate whether or not you've manually edited any data directly in the database
-- No.

-----
Screenshots

![ldap1](https://user-images.githubusercontent.com/38566019/66860889-d7dd5f00-ef42-11e9-8393-f051aa54f003.JPG)
![ldap2](https://user-images.githubusercontent.com/38566019/66860891-d7dd5f00-ef42-11e9-90c4-08a5a261c8d5.JPG)

-----

- If a stacktrace is provided in the error, include that too.
[2019-10-15 09:13:43] production.ERROR: ldap_search(): Search: Invalid DN syntax {"userId":1,"email":"*******@***.com","exception":"[object] (ErrorException(code: 0): ldap_search(): Search: Invalid DN syntax at /var/www/snipe-it/app/Models/Ldap.php:265)
[stacktrace]
#0 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(2, 'ldap_search(): ...', '/var/www/snipe-...', 265, Array)
#1 /var/www/snipe-it/app/Models/Ldap.php(265): ldap_search(Resource id #67, 'ou=india,ou-agr...', '(&(objectCatego...')
#2 /var/www/snipe-it/app/Console/Commands/LdapSync.php(128): App\\Models\\Ldap::findLdapUsers('ou=india,ou-agr...')
#3 [internal function]: App\\Console\\Commands\\LdapSync->handle()
#4 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#5 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#6 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#7 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#8 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\\Container\\Container->call(Array)
#9 /var/www/snipe-it/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArrayInput), Object(Illuminate\\Console\\OutputStyle))
#10 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArrayInput), Object(Illuminate\\Console\\OutputStyle))
#11 /var/www/snipe-it/vendor/symfony/console/Application.php(960): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArrayInput), Object(Symfony\\Component\\Console\\Output\\BufferedOutput))
#12 /var/www/snipe-it/vendor/symfony/console/Application.php(255): Symfony\\Component\\Console\\Application->doRunCommand(Object(App\\Console\\Commands\\LdapSync), Object(Symfony\\Component\\Console\\Input\\ArrayInput), Object(Symfony\\Component\\Console\\Output\\BufferedOutput))
#13 /var/www/snipe-it/vendor/symfony/console/Application.php(148): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArrayInput), Object(Symfony\\Component\\Console\\Output\\BufferedOutput))
#14 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Console/Application.php(88): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArrayInput), Object(Symfony\\Component\\Console\\Output\\BufferedOutput))
#15 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Console/Application.php(177): Illuminate\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArrayInput), Object(Symfony\\Component\\Console\\Output\\BufferedOutput))
#16 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(249): Illuminate\\Console\\Application->call('snipeit:ldap-sy...', Object(Illuminate\\Support\\Collection), NULL)
#17 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(221): Illuminate\\Foundation\\Console\\Kernel->call('snipeit:ldap-sy...', Array)
#18 /var/www/snipe-it/app/Http/Controllers/UsersController.php(875): Illuminate\\Support\\Facades\\Facade::__callStatic('call', Array)
#19 [internal function]: App\\Http\\Controllers\\UsersController->postLDAP(Object(Illuminate\\Http\\Request))
#20 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): call_user_func_array(Array, Array)
#21 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction('postLDAP', Array)
#22 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Route.php(212): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(App\\Http\\Controllers\\UsersController), 'postLDAP')
#23 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Route.php(169): Illuminate\\Routing\\Route->runController()
#24 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Router.php(658): Illuminate\\Routing\\Route->run()
#25 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(30): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#26 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Auth/Middleware/Authenticate.php(43): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#27 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Illuminate\\Auth\\Middleware\\Authenticate->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#28 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#29 /var/www/snipe-it/vendor/laravel/passport/src/Http/Middleware/CreateFreshApiToken.php(49): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#30 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Laravel\\Passport\\Http\\Middleware\\CreateFreshApiToken->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#31 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#32 /var/www/snipe-it/app/Http/Middleware/CheckForTwoFactor.php(49): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#33 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\CheckForTwoFactor->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#34 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#35 /var/www/snipe-it/app/Http/Middleware/CheckLocale.php(42): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#36 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\CheckLocale->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#37 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#38 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(68): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#39 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#40 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#41 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#42 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#43 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#44 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(66): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#45 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#46 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#47 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#48 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Router.php(660): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#49 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Router.php(635): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#50 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Router.php(601): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#51 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Router.php(590): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#52 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(176): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#53 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(30): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#54 /var/www/snipe-it/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(65): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#55 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Barryvdh\\Debugbar\\Middleware\\InjectDebugbar->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#56 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#57 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(30): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#58 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#59 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#60 /var/www/snipe-it/app/Http/Middleware/CheckForDebug.php(25): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#61 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\CheckForDebug->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#62 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#63 /var/www/snipe-it/app/Http/Middleware/CheckForSetup.php(27): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#64 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\CheckForSetup->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#65 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#66 /var/www/snipe-it/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#67 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Fideloper\\Proxy\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#68 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#69 /var/www/snipe-it/app/Http/Middleware/NosniffGuard.php(17): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#70 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\NosniffGuard->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#71 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#72 /var/www/snipe-it/app/Http/Middleware/ContentSecurityPolicyHeader.php(18): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#73 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\ContentSecurityPolicyHeader->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#74 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#75 /var/www/snipe-it/app/Http/Middleware/ReferrerPolicyHeader.php(17): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#76 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\ReferrerPolicyHeader->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#77 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#78 /var/www/snipe-it/app/Http/Middleware/XssProtectHeader.php(18): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#79 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\XssProtectHeader->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#80 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#81 /var/www/snipe-it/app/Http/Middleware/FrameGuard.php(17): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#82 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): App\\Http\\Middleware\\FrameGuard->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#83 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#84 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#85 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#86 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#87 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(63): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#88 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Illuminate\\Session\\Middleware\\StartSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#89 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#90 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(46): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#91 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#92 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#93 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#94 /var/www/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(151): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))

**Comment by prodweb1** [comment]
same issue with 4.7.5

**Comment by voegel916** [comment]
Not exactly a fix, but I reinstalled Snipe-It on another VM, this time using version 4.7.7. Copied the same LDAP settings I used before, and the LDAP sync worked perfectly. Possibly a bug? It could have been something messed up during installation or configuration, though I'm not sure what since the settings are limited. Either way, it's working now, and I needed to update anyway :)

**Comment by HinchK** [comment]
Just a head's up for anyone encountering the problem related to `Invalid DN Syntax` with regard to Snipe-IT LDAP Connections _that were previously working_ in a previous release, but appear to not be working > v4.6.0.

Please check the `location.ldap_ou` settings across your locations, and from the CLI, try the following command:

`php artisan snipeit:ldap-sync --summary --base_dn "ou=Users,dc=your,dc=domain,dc=com"`

Stay safe friends!

**Comment by bruce2004x** [comment]
Dear HinchK, where is located the file that contents the location.ldap_ou settings? I run the command you said and works fine, but when I try to do the sync via GUI it fails. I have tried everithing without good results.

Thanks in advance.

**Comment by HinchK** [comment]
Great question @bruce2004x - I realize now my comment may have been as clear as mud!

TLDR `location.ldap_ou` refers to the `ldap_ou` column of your Snipe-IT database's `location` table. 

**
![Uploading image.png…]
**

When you update or create a new location, there is a field "LDAP Search OU."  

The contents of that textbox are added onto the BASE-DN right before syncing; if you were to select a location from the dropdown and click "sync" - OR if you select no locations and click sync...Snipe-IT will do it's best to put your users in the right locations.

MOST of the time - if you get `invalid BASE_DN`, but `php artisan snipeit:ldap-sync --summary` works - the likely culprit is a typo in a location's ldap_ou field.

Checking for weird values from the `mysql>` console:

```sql
select ldap_ou 
from snipeitmaster.locations 
where ldap_ou is not null;
```

**Comment by bruce2004x** [comment]
Dear HinchK, you were right, I found this:

MariaDB [snipeit]> select ldap_ou from locations where ldap_ou is not null;
+---------+
| ldap_ou |
+---------+
| Ecuador |
+---------+
1 row in set (0.00 sec)

Where "Ecuador" is just the name of one of my OU's in AD. My question now is, what should I put there? just delete that value?

Thanks in advance.

**Comment by bruce2004x** [comment]
Friend,

I just run:

 delete from locations where locations.ldap_ou = "record";

where "record" was the value found in the table and you know what? IT WORKED!!!!

Thanks million, I'm sure that this workaround might help others.

Best,

Eduardo from Colombia
